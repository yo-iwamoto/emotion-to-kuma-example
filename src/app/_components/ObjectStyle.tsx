'use client';

import { css } from '@emotion/react';

export function ObjectStyle() {
  return <p css={styles.text}>ObjectStyle</p>;
}

const styles = {
  text: css({
    color: 'blue',
    fontSize: 40,
  }),
};
