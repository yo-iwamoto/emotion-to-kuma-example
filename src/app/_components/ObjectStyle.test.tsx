import { ObjectStyle } from './ObjectStyle';
import { render } from '@testing-library/react';

describe('ObjectStyle', () => {
  it('renders unchanged', () => {
    const { asFragment } = render(<ObjectStyle />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('has color style', () => {
    const { getByText } = render(<ObjectStyle />);

    expect(getByText('ObjectStyle')).toHaveStyleRule('color', 'blue');
  });
});
