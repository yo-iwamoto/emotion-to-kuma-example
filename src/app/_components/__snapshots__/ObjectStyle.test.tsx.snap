// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ObjectStyle > renders unchanged 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: blue;
  font-size: 40px;
}

<p
    class="emotion-0"
  >
    ObjectStyle
  </p>
</DocumentFragment>
`;
